test/parallel/test-crypto-timing-safe-equal.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+
+if (!common.hasCrypto) {
+  common.skip('missing crypto');
+  return;
+}
+
+const crypto = require('crypto');
+
+assert.strictEqual(
+  crypto.timingSafeEqual(Buffer.from('foo'), Buffer.from('foo')),
+  true,
+  'should consider equal strings to be equal'
+);
+
+assert.strictEqual(
+  crypto.timingSafeEqual(Buffer.from('foo'), Buffer.from('bar')),
+  false,
+  'should consider unequal strings to be unequal'
+);
+
+assert.throws(function() {
+  crypto.timingSafeEqual(Buffer.from([1, 2, 3]), Buffer.from([1, 2]));
+}, 'should throw when given buffers with different lengths');
+
+assert.throws(function() {
+  crypto.timingSafeEqual('not a buffer', Buffer.from([1, 2]));
+}, 'should throw if the first argument is not a buffer');
+
+assert.throws(function() {
+  crypto.timingSafeEqual(Buffer.from([1, 2]), 'not a buffer');
+}, 'should throw if the second argument is not a buffer');
+
+function runBenchmark(compareFunc, bufferA, bufferB) {
+  const startTime = process.hrtime();
+  const result = compareFunc(bufferA, bufferB);
+  const endTime = process.hrtime(startTime);
+
+  // Ensure that the result of the function call gets used, so that it doesn't
+  // get discarded due to engine optimizations.
+  assert.strictEqual(result, bufferA.equals(bufferB));
+  return endTime[1];
```

Comment:
If you only check `endTime[1]`, you may want to add an `assert.strictEqual(endTime[0], 0);` here (or just take `endTime[0]` into account).